sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/StorageResourceNamer.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.test.shared;
+
+import com.azure.core.test.TestMode;
+import com.azure.core.test.models.RecordedData;
+import com.azure.core.test.utils.TestResourceNamer;
+
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.zip.CRC32;
+
+public class StorageResourceNamer {
+    private final TestResourceNamer testResourceNamer;
+    private final String resourcePrefix;
+
+    public StorageResourceNamer(String testName, TestMode testMode, RecordedData recordedData) {
+        Objects.requireNonNull(testName);
+        Objects.requireNonNull(testMode);
+        Objects.requireNonNull(recordedData);
+        resourcePrefix = getCrc32(testName);
```

Comment:
Any reason to change the resource prefix to CRC32 instead of using the test name as previously was being done

Reply:
"test bla" and "test bla fail" collide at cleanup. If run in parallel this creates race conditions.
I was experimenting with just uuid, but this one is shorter (we'll revisit that choice if collisions are problematic, but I don't recall this being a problem in python where they crc32)

Reply:
I've seen plenty of that when running blobs in parallel. Which yielded this workaround https://github.com/Azure/azure-sdk-for-java/blob/9ee4c8904e6f1d3f9c3fafd51bc6ae3a4b4d40a2/sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy#L172-L176
This should enable parallelization in all modes.